DataOnion/AuthExtensions.cs

```diff
@@ -0,0 +1,84 @@
+using Microsoft.Extensions.DependencyInjection;
+using DataOnion.Auth;
+using StackExchange.Redis;
+
+namespace DataOnion
+{
+    public interface IFluentAuthOnion
+    {
+        IFluentAuthOnion ConfigureSlidingExpiration<T>(
+            TimeSpan expiration,
+            TimeSpan? absoluteExpiration,
+            string authPrefix,
+            Func<HashEntry[], T> makeFromHash,
+            string expirationKey = "expiration"
+        )
+            where T : class, IAuthStorable<T>;
+
+        IFluentAuthOnion ConfigureRedis(string connectionString);
```

Comment:
i want to make it so the consumer of the library has to pick a strategy before configuring redis that way the fluent API makes more sense and has to change less if we ever use another technology as the caching layer

then i want to add a function that can be called after the configure strategy method but before the configure redis method that can be called `WithCustomStrategy` or something like that